### ISSUE TITLE
Nvidia NSight Tegra Android Visual Studio project generation

### ISSUE THREAD

**Issue by Muzza** [bkaradzic/GENie#158]
NSight-Tegra is a plug-in for Visual Studio that allows Android java/c++ projects to be built and deployed to any Android device. 
See https://developer.nvidia.com/nvidia-nsight-tegra for more info

I've extended GENie to be able to output visual studio projects targeting NSight-Tegra, but it does require a number of additional options for things like setting Android Min/Target SDKs and for handling java paths, jar paths, proguard, assets, and more. It also occasionally differs from standard Visual Studio project files - here is one example:

```lua
if cfg.platform == "TegraAndroid" then
	_p(3,'<OptimizationLevel>%s</OptimizationLevel>',androidOptimisation(cfg))
else
	_p(3,'<Optimization>%s</Optimization>',optimisation(cfg))
end
```

This is all extremely useful for me, but it may be a minority interest item for others that just clutters up GENie with extra code. So I'm asking here, is there any interest in this feature being added to the main branch, or should I just leave it?

**Comment by bkaradzic** [comment]
It would be good if this could be configured outside of GENie source. I've started adding `premake.vstudio` options, and I intend to go that route for future additions rather than trying to figure out high-level options.

**Comment by Muzza** [comment]
I'm not sure what you mean by 'outside of GENie source'.

**Comment by bkaradzic** [comment]
See this: https://github.com/bkaradzic/bx/blob/master/scripts/toolchain.lua

Since Lua has everything global, you could configure at lot of things without changing source of GENie.

**Comment by Muzza** [comment]
I've looked, and either I'm missing something or GENie would have to allow more things to be overriden before I could make that work.

In the code example I posted above, the only way I can see that I could customise the Optimization tag would be to make an entirely new premake.vs2010_vcxproj function, which results in masses of code duplication.

**Comment by bkaradzic** [comment]
> GENie would have to allow more things to be overriden

Yes. In this particular case it might not be possible because you want to depend on configuration too, but probably there are some VS settings that could be exposed via `premake.vstudio` that would make easy to extend with newer platforms and toolchains.
